examples/tokens/ERC1155.vy

```diff
@@ -0,0 +1,267 @@
+# function safeTransferFrom(address _from, address _to, uint256 _id, uint256 _value, bytes calldata _data) external;
+@external
+def safeTransferFrom(
+    _from: address,
+    _to: address,
+    _id: uint256,
+    _value: uint256,
+    _data: Bytes[256]
+  ):
+    assert _from == msg.sender or (self.operators[_from])[msg.sender]
+    assert _to != ZERO_ADDRESS
+    assert self._balanceOf[_from][_id] >= _value
+
+    if _to.is_contract:
+        returnValue: bytes32 = ERC1155TokenReceiver(_to).onERC1155Received(msg.sender, _from, _id, _value, _data)
+        assert returnValue == method_id("onERC1155Received(address,address,uint256,uint256,bytes)", output_type=bytes32)
+
+    self._balanceOf[_from][_id] -= _value
+    self._balanceOf[_to][_id] += _value
+    log TransferSingle(msg.sender, _from, _to, _id, _value)
+
+
+# @notice Transfers `_values` amount(s) of `_ids` from the `_from` address to the `_to` address specified (with safety call).
+# @dev Caller must be approved to manage the tokens being transferred out of the `_from` account (see "Approval" section of the standard).
+#      MUST revert if `_to` is the zero address.
+#      MUST revert if length of `_ids` is not the same as length of `_values`.
+#      MUST revert if any of the balance(s) of the holder(s) for token(s) in `_ids` is lower than the respective amount(s) in `_values` sent to the recipient.
+#      MUST revert on any other error.
+#      MUST emit `TransferSingle` or `TransferBatch` event(s) such that all the balance changes are reflected (see "Safe Transfer Rules" section of the standard).
+#      Balance changes and events MUST follow the ordering of the arrays (_ids[0]/_values[0] before _ids[1]/_values[1], etc).
+#      After the above conditions for the transfer(s) in the batch are met, this function MUST check if `_to` is a smart contract (e.g. code size > 0). If so, it MUST call the relevant `ERC1155TokenReceiver` hook(s) on `_to` and act appropriately (see "Safe Transfer Rules" section of the standard).
+# @param _from    Source address
+# @param _to      Target address
+# @param _ids     IDs of each token type (order and length must match _values array)
+# @param _values  Transfer amounts per token type (order and length must match _ids array)
+# @param _data    Additional data with no specified format, MUST be sent unaltered in call to the `ERC1155TokenReceiver` hook(s) on `_to`
+# function safeBatchTransferFrom(address _from, address _to, uint256[] calldata _ids, uint256[] calldata _values, bytes calldata _data) external;
+@external
+def safeBatchTransferFrom(
+    _from: address,
+    _to: address,
+    _ids: uint256[BATCH_SIZE],
+    _values: uint256[BATCH_SIZE],
+    _data: Bytes[256]
+  ):
+    assert _from == msg.sender or (self.operators[_from])[msg.sender]
+    assert _to != ZERO_ADDRESS
+    #assert len(_ids) == len(_values)
+
+    for i in range(BATCH_SIZE):
```

Comment:
```suggestion
    for i in range(min(len(_ids), BATCH_SIZE)):
```

Reply:
Note: this is because Vyper does not allow unbounded loops, but will allow `min(var, const)` expressions as it has a defined upper bound

Reply:
> Note: this is because Vyper does not allow unbounded loops, but will allow `min(var, const)` expressions as it has a defined upper bound

oh so I think this for loop is limited by `BATCH_SIZE`.  I also saw your comment here #1047 . could you help me merge pull requests bro. or give me some suggestions. thank bro @fubuloubu